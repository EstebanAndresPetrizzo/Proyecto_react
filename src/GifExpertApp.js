import React,{useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () => {
    //const categories = ['PrimerElemento','SegundoElemento','TercerElemento'];//se utiliza si fuera una lista que no cambiara
    const [categories, setCategories] = useState(['Batman']);
    
 /*   const handleAdd = () =>{
        //setCategories([...categories,'OtroElemento']);
                //los tres puntos se utilizan para extraer lo que esta en la variable
                //estas quedaran primero, si se desea agregar al principio
                //solo invertimos el orden

        //otra forma
        setCategories(cats => [...categories,'OtroElemento']);

    }
*/
    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>

            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }    
            </ol>
        </>//ol - Lista ordenada
    );
}