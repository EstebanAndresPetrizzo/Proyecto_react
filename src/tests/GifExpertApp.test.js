import { shallow } from "enzyme"
import React from 'react'
import '@testing-library/jest-dom';

import {GifExpertApp} from '../GifExpertApp';


describe('Prueba en <GitExpertApp />', () =>{

    let wrapper = shallow(<GifExpertApp /> );
    beforeEach(() => {
        wrapper = shallow(<GifExpertApp />);
    });


    test('debe mostrar un <GitExpertApp />', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar una lista de categories />', () => {
        const categories = ['Batman', 'One Punch'];
        wrapper = shallow(<GifExpertApp dafaultCategories = {categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
})