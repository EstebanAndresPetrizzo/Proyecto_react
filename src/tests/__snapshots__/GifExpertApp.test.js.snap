// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GitExpertApp /> debe mostrar un <GitExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prueba en <GitExpertApp /> debe mostrar una lista de categories /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
